### ISSUE TITLE
[data grid] large whitespace area appears in filtered results

### ISSUE THREAD

**Issue by rolule** [mui/mui-x#5470]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Current behavior 😯

A large whitespace area appears before the first row of the filtered results and not all filtered results are shown:

![Screen Shot 2022-07-12 at 18 13 47](https://user-images.githubusercontent.com/73437576/178541949-75685238-8478-4ac8-af95-a6138b2c2459.png)
In the screenshot, 10 filtered results are indicated in the pagination, but only the last 6 items are shown (compare to expected behavior).

 This issue seems to only appear with a large amount of cells with similar content and when using `getRowHeight={() => "auto"}`.

### Expected behavior 🤔

The rows should adapt their size based on their content and the whitespace area should not show up when filtering.

![Screen Shot 2022-07-12 at 18 18 26](https://user-images.githubusercontent.com/73437576/178542762-c3bfbe20-0eb0-4ae7-b1bf-5ee4dd6ecac0.png)

### Steps to reproduce 🕹

Link to live example: https://codesandbox.io/s/data-grid-filter-issue-97cpny?file=/src/App.tsx

Steps:
1. Open the CodeSandbox link above
2. Click on the context menu of the "name" row and click "filter"
3. Type in the word "bar" in the filter input field
4. Click away the filter menu to see the filter results more easily

### Context 🔦

We noticed this issue while testing with a dataset that contains a lot of table rows that have a column cell containing the word "test". When filtering this column for "test", the whitespace bug appeared.

### Your environment 🌎

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
  Used Browser: Brave / Chrome [Version 1.40.113 Chromium: 103.0.5060.114 (Official Build) (arm64)]
  
  System:
    OS: macOS 12.2
  Binaries:
    Node: 16.15.0 - ~/.nvm/versions/node/v16.15.0/bin/node
    Yarn: 1.22.15 - ~/.nvm/versions/node/v16.15.0/bin/yarn
    npm: 8.5.5 - ~/.nvm/versions/node/v16.15.0/bin/npm
  Browsers:
    Chrome: 103.0.5060.114
    Edge: Not Found
    Firefox: 101.0.1
    Safari: 15.3
  npmPackages:
    @emotion/react: 11.9.3 => 11.9.3 
    @emotion/styled:  11.9.3 
    @mui/base:  5.0.0-alpha.88 
    @mui/icons-material: 5.8.4 => 5.8.4 
    @mui/lab: 5.0.0-alpha.89 => 5.0.0-alpha.89 
    @mui/material: 5.8.7 => 5.8.7 
    @mui/private-theming:  5.8.6 
    @mui/styled-engine:  5.8.7 
    @mui/system: 5.8.7 => 5.8.7 
    @mui/types:  7.1.4 
    @mui/utils:  5.8.6 
    @mui/x-data-grid: 5.13.0 => 5.13.0 
    @mui/x-date-pickers:  5.0.0-alpha.1 
    @types/react: 17.0.45 => 17.0.45 
    react: 17.0.2 => 17.0.2 
    react-dom: 17.0.2 => 17.0.2 
    styled-components: 5.3.5 => 5.3.5 
    typescript: 4.7.4 => 4.7.4
```
</details>

**Comment by m4theushw** [comment]
We already fixed this bug in #5315. It will be released this week or next week.

Proof: https://codesandbox.io/s/data-grid-filter-issue-forked-zrb2rn?file=/src/App.tsx

**Comment by rolule** [comment]
Very nice, thank you. Sorry that I could't find the existing issue.
